pkg/symnotify/symnotify.go

```diff
@@ -49,56 +50,67 @@ func (w *Watcher) Event() (e Event, err error) {
 	log.V(3).Info("event", "path", e.Name, "operation", e.Op.String())
 	switch {
 	case e.Op == Create:
-		if info, err := os.Lstat(e.Name); err == nil {
+		var info os.FileInfo
+		if info, err = os.Lstat(e.Name); err == nil {
 			if isSymlink(info) || info.IsDir() {
-				_ = w.Add(e.Name)
+				err = w.Add(e.Name)
 			}
 		}
 	case e.Op == Remove:
-		_ = w.watcher.Remove(e.Name)
+		err = w.watcher.Remove(e.Name)
 	case e.Op == Chmod || e.Op == Rename:
-		if info, err := os.Lstat(e.Name); err == nil {
+		var info os.FileInfo
+		if info, err = os.Lstat(e.Name); err == nil {
 			if isSymlink(info) {
 				// Symlink target may have changed.
-				_ = w.watcher.Remove(e.Name)
-				_ = w.watcher.Add(e.Name)
+				err = w.watcher.Remove(e.Name)
+				err = w.watcher.Add(e.Name)
 			}
 		}
 	}
+	if err != nil {
+		log.V(3).Error(err, "Error retrieving event", "path", e.Name, "operation", e.Op.String())
+	}
 	return e, nil
 }
 
+// Remove name from watcher
+func (w *Watcher) Remove(name string) error {
+	log.V(3).Info("stop watching", "path", name)
+	return w.watcher.Remove(name)
+}
+
 // Add a new directory, file or symlink to be watched.
 func (w *Watcher) Add(name string) (err error) {
 	log.V(3).Info("start watching", "path", name)
-	if err := w.watcher.Add(name); err != nil {
+	if err = w.watcher.Add(name); err != nil {
 		log.Error(err, "error watching", "path", name)
 		return err
 	}
 	// If name is a directory, scan for existing symlinks and sub-directories to watch.
-	if infos, err := ioutil.ReadDir(name); err == nil {
+	var infos []fs.FileInfo
+	if infos, err = ioutil.ReadDir(name); err == nil {
 		for _, info := range infos {
+			log.V(3).Info("Checking path for more files to watch", "name", name, "subpath", info.Name())
 			newName := filepath.Join(name, info.Name())
 			switch {
 			case info.IsDir():
-				return w.Add(newName)
+				if err := w.Add(newName); err != nil {
```

Comment:
do you want to redefine `err` here? earlier function was returning here. perhaps you want `(err error)` to be updated here.

Reply:
accumulation of errors would require something like https://github.com/hashicorp/go-multierror package. need to use this pkg for accessing the individual errors also. Ultimately i think these errors are for printing. so why not print these here itself.

my question was about `err := w.Add(newName)` vs `err = w.Add(newName)` since `err` is already declared. using `:=` shadows the earlier defined variable.

Reply:
@vimalk78 I can modify i to satisfy your query but given we don't do anything other then log the value seems trivial.  I likely could return an accumulated error without introducing the new package. Do you think:
* we should accumulate and return?
* We should modify the level from 3 to 0?

Reply:
We should modify the level from 3 to 0

Reply:
I remembered why i want to shadow the error.  The only way to get that error is if it is returned from the directory call which we want to return.  If it errors from the loop we don't want to return those errors

Reply:
then a different variable name would have been better